src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -302,6 +323,9 @@ internal static string GetRelativeDirectory(string path, string scriptRoot)
 
         private async Task ScheduleRestartAsync(bool shutdown)
         {
+            // will return when the count reaches 0 or after 1 minute
+            _cde.Wait(60000);
```

Comment:
We really don't want to block a thread on this. Unfortunately there isn't an async version of this class. Might be simpler for us to to implement this ourselves via a simple syncronized counter, and use of our Utility.DelayAsync helper, to delay (without blocking) until the counter reaches 0.

Reply:
There's no need to wait here. If we reach this and notifications are suspended, something didn't work as expected. We don't want to block threads and have them all fire here when the notifications are resumed.

If restart/notifications are suspended, we should just return here.

Reply:
@fabiocav, I do not quiet understand "notifications are suspended". We are processing a file change, initiate restart flow and wait right before the restart until counter is 0. Can you please clarify?

Reply:
@fabiocav, does "suspending notification" mean Initialze/Stop FileWatchers?

Reply:
You shouldn't have to actually stop filewatchers. You can just track if a restart was requested during the suspension, and if it was when we resume, you'd invoke it then.